vaultclient/src/main/java/com/scality/vaultclient/dto/GetAccountRequestMarshaller.java

```diff
@@ -0,0 +1,82 @@
+import com.amazonaws.protocol.MarshallingInfo;
+import com.amazonaws.protocol.MarshallingType;
+import com.amazonaws.protocol.ProtocolRequestMarshaller;
+import com.amazonaws.util.StringUtils;
+import com.scality.vaultclient.services.VaultClientException;
+
+/**
+ * The type Get account request marshaller extends GenericRequestMarshaller to marshall Get Account Requests.
+ */
+public class GetAccountRequestMarshaller extends GenericRequestMarshaller<GetAccountRequestDTO>{
+
+    private static final MarshallingInfo<String> ACCOUNT_NAME_BINDING = MarshallingInfo.builder(MarshallingType.STRING).marshallLocation(MarshallLocation.PAYLOAD)
+            .marshallLocationName("accountName").build();
+    private static final MarshallingInfo<String> ACCOUNT_ID_BINDING = MarshallingInfo.builder(MarshallingType.STRING).marshallLocation(MarshallLocation.PAYLOAD)
+            .marshallLocationName("accountId").build();
+    private static final MarshallingInfo<String> ACCOUNT_ARN_BINDING = MarshallingInfo.builder(MarshallingType.STRING).marshallLocation(MarshallLocation.PAYLOAD)
+            .marshallLocationName("accountArn").build();
+    private static final MarshallingInfo<String> CANONICAL_ID_BINDING = MarshallingInfo.builder(MarshallingType.STRING).marshallLocation(MarshallLocation.PAYLOAD)
+            .marshallLocationName("canonicalId").build();
+    private static final MarshallingInfo<String> EMAIL_ADDRESS_BINDING = MarshallingInfo.builder(MarshallingType.STRING).marshallLocation(MarshallLocation.PAYLOAD)
+            .marshallLocationName("emailAddress").build();
+
+
+    private static final String GET_ACCOUNT_ACTION = "GetAccount";
+
+    private static final GetAccountRequestMarshaller instance = new GetAccountRequestMarshaller();
+
+    public static GetAccountRequestMarshaller getInstance() {
+        return instance;
+    }
+
+    /**
+     * Binds protocolMarshaller with getAccountRequestDTO parameters to marshall.
+     *
+     * @param getAccountRequestDTO         the request dto of type GetAccountRequestDTO
+     * @param protocolMarshaller the protocol marshaller
+     */
+    @Override
+    public void marshall(GetAccountRequestDTO getAccountRequestDTO, ProtocolRequestMarshaller<GetAccountRequestDTO> protocolMarshaller) {
+
+        super.marshall(getAccountRequestDTO, protocolMarshaller, GET_ACCOUNT_ACTION);
+
+        try {
+            if (StringUtils.isNullOrEmpty(getAccountRequestDTO.getAccountName())
+                    && StringUtils.isNullOrEmpty(getAccountRequestDTO.getAccountId())
+                    && StringUtils.isNullOrEmpty(getAccountRequestDTO.getAccountArn())
+                    && StringUtils.isNullOrEmpty(getAccountRequestDTO.getCanonicalId())
+                    && StringUtils.isNullOrEmpty(getAccountRequestDTO.getEmailAddress()))
+             {
+                throw new VaultClientException("At least one of account-name, account-id, email, account-arn or canonical-id needs to be specified");
```

Comment:
As per the code changes in vault only 1 should be specified.
Reference: https://github.com/scality/Vault/blob/development/7.10/lib/Validator/getAccount.js#L27

Reply:
Correct. As this is only a client I am not validating if multiple parameters are provided. This is only a validation to verify if no parameter is passed. This is based on `vaultclient` code
Reference: https://github.com/scality/vaultclient/blob/7205377ec11aed8e682017ab7cb9d066eaf8cd90/lib/IAMClient.js#L407

Reply:
I would suggest, eventually, we need to validate if multiple parameters are provided. Would definitely help save some network expenses.
If you are not doing it now, worth creating a ticket for it.

This is how we do it in Vaultclient: https://github.com/scality/vaultclient/blob/development/7.10/lib/IAMClient.js#L444

Reply:
@anurag4DSB  Fixed in 98ba3dcae6bf39c87201fb20481e7cad1f79e1f2